src/js/components/Widgets/ShowMoreButtons.jsx

```diff
@@ -5,12 +5,52 @@ import PropTypes from 'prop-types';
 import React from 'react';
 import styled from 'styled-components';
 import { renderLog } from '../../common/utils/logging';
-
+import lookupPageNameAndPageTypeDict from '../../utils/lookupPageNameAndPageTypeDict';
+import TagManager from 'react-gtm-module';
+import VoterStore from '../../stores/VoterStore';
 
 class ShowMoreButtons extends React.Component {
-  render () {
-    renderLog('ShowMoreButtons');  // Set LOG_RENDER_EVENTS to log all renders
-    const { classes, showLessCustomText, showMoreButtonsLink, showMoreButtonWasClicked, showMoreCustomText, showMoreId } = this.props;
+  
+  constructor(props) {
+    super(props);
+    this.handleShowMoreClick = this.handleShowMoreClick.bind(this);
+    this.pushDataLayer = this.pushDataLayer.bind(this);
+  }
+
+  handleShowMoreClick = () => {
+    const { showMoreId, showMoreButtonWasClicked, trackInGTM, showMoreButtonsLink } = this.props;
+    const { location: { pathname: currentPathname } } = window; // Get path here
+    //if (trackInGTM) {
+      //console.log('click');
+      const eventName = showMoreButtonWasClicked ? 'showLessButtonClick' : 'showMoreButtonClick';
+      this.pushDataLayer(eventName, showMoreId, currentPathname); // Use currentPathname
+    //} else {
+      //console.log('Show More/Less clicked (not tracked):', showMoreId);
+    //}
+    showMoreButtonsLink();
```

Comment:
Even though the prop `showMoreButtonsLink` is required, I think it is a good practice to wrap calls to prop functions with an if like this: 
```
if (showMoreButtonsLink) {
  showMoreButtonsLink();
}
```